met-web/src/components/engagement/form/ActionContext.tsx

```diff
@@ -83,59 +74,68 @@ export const ActionProvider = ({ children }: { children: JSX.Element }) => {
     const [savedBannerImageFileName, setSavedBannerImageFileName] = useState('');
     const [contentTabs, setContentTabs] = useState<EngagementContent[]>([createDefaultEngagementContent()]);
     const isCreate = window.location.pathname.includes(CREATE);
+    const noRouteLoaders = { engagement: 'engagement', content: 'content', metadata: 'metadata', taxa: 'taxa' };
+    const { engagement } = !isCreate
+        ? (useRouteLoaderData('single-engagement') as { engagement: Promise<Engagement> })
+        : noRouteLoaders;
+    const { content } = !isCreate
+        ? (useRouteLoaderData('single-engagement') as { content: Promise<EngagementContent[]> })
+        : noRouteLoaders;
+    const { metadata } = !isCreate
+        ? (useRouteLoaderData('single-engagement') as { metadata: Promise<EngagementMetadata[]> })
+        : noRouteLoaders;
+    const { taxa } = !isCreate
```

Comment:
It needs to be destructured due to the way the loader is structured, outputting different values as object properties. Of course, this could be changed.